src/main/java/org/mockito/Mockito.java

```diff
@@ -1772,6 +1794,31 @@ public static MockingDetails mockingDetails(Object toInspect) {
     }
 
     /**
+     * Downcasts mock object to a specific type. This is helpful for Java 8 where
+     * <code>mock(Predicate.class)</code> cannot provide a compiler-friendly value
+     * for a variable of, say, <code>Predicate&lt;String&gt;</code>. If you
+     * wrap the call to <code>mock</code> in <code>typed(...)</code>, then you will not
+     * get compiler warnings.
+     * <p>
+     *
+     * <pre class="code"><code class="java">
+     *   Foo&lt;Bar&gt; mock = typed(mock(Foo.class));
+     * </code></pre>
+     *
+     * <p>
+     * <em>Warning:</em> this method could be accidentally used to downcast to
+     * a subclass of the mock, resulting in a <code>ClasCastException</code>
```

Comment:
Is there a test that shows this behavior? Or maybe include a direct example pointing out when this is an issue.

Reply:
I can add a test for it...

I will.

Essentially the constraint within the `<M, T extends M>` in the `typed` method means that the types must be related with the cast being to a more specific type than the mock's raw type from the class. However, if you tried to force a cast of `mock(List.class)` to `ArrayList`, it wouldn't work.

Give me a moment, and I'll add a test.

Reply:
There it is.